lib/forest_liana/bootstraper.rb

```diff
@@ -178,9 +195,36 @@ def create_apimap
       end
     end
 
+    def load_apimap
+      if File.exists?(File.join(Rails.root, 'forestadmin-schema.json'))
+        begin
+          apimap = File.read(File.join(Rails.root, 'forestadmin-schema.json'))
+          apimap = JSON.parse(apimap)
+          ForestLiana.meta = apimap['meta']
+          ForestLiana.apimap = ForestLiana::Model::Collection.from_json(apimap['collections'])
+        rescue JSON::JSONError
+          FOREST_LOGGER.error "File forestadmin-schema.json does not appear to be valid json."
+        end
+      else
+        FOREST_LOGGER.error "File forestadmin-schema.json does not exist."\
+          "Make sure to deploy this file in your production environment."
+      end
+    end
+
     def create_apimap_json
       File.open(File.join(Rails.root, 'forestadmin-schema.json'), 'w') do |f|
-        f.puts JSON.pretty_generate(ForestLiana.apimap.as_json)
+        collections = ForestLiana.apimap.as_json
+        collections = collections.map do |collection|
+          collection['fields'].sort { |field1, field2| [field1['field'], field1['type']] <=> [field2['field'], field2['type']] }
+          collection['fields'] = collection['fields'].map { |field| field.sort.to_h }
+          collection.sort.to_h
+        end
+        collections.sort { |collection1, collection2| collection1['name'] <=> collection2['name'] }
+
+        f.puts JSON.pretty_generate({
+          collections: collections,
+          meta: ForestLiana.meta
+        })
```

Comment:
Something like this:
Here is what we have so far:
```json
"collections": [{
  "name": "Company",
  "name_old": "companies",
  "is_read_only": false,
  "is_searchable": true,
  "fields": [{
    "field": "id",
    "type": "Number"
  }, {
      "field": "name",
      "type": "String"
  }, {
    "field": "industry",
    "type": "String"
  }, {
    "field": "headquarter",
    "type": "String"
  }, {  
    "field": "status",
    "type": "Enum",
    "enums": [
      "signed_up",
      "pending",
      "approved",
      "rejected",
      "live"
    ],
  }, {
    "field": "description",
    "type": "String"
  }, {
    "field": "created_at",
    "type": "Date"
  }, {
    "field": "updated_at",
    "type": "Date"
  }, {
    "field": "certificate_of_incorporation_id",
    "type": "String"
  }, {
    "field": "proof_of_address_id",
    "type": "String"
  }, {
    "field": "bank_statement_id",
    "type": "String"
  }, {
    "field": "passport_id",
    "type": "String"
  }, {
    "field": "received_transactions",
    "type": ["Number"],
    "inverseOf": null,
    "is-filterable": false,
    "reference": "Transaction.id"
  }, {
    "field": "emitted_transactions",
    "type": ["Number"],
    "inverseOf": null,
    "is-filterable": false,
    "reference": "Transaction.id"
  }],
  "segments": [],
  "actions": [{
    "name": "Mark as Live"
  }, {
    "name": "Add new transaction",
    "fields": [{
      "field": "Beneficiary company",
      "description": "Name of the company who will receive the transaction.",
      "reference": "Company.id",
      "position": 0
    }, {
      "field": "Amount",
      "type": "Number",
      "position": 1
    }]
  }, {
    "name": "Upload Legal Docs",
    "type": "single",
    "fields": [{
      "field": "Certificate of Incorporation",
      "description": "The legal document relating to the formation of a company or corporation.",
      "type": "File",
      "isRequired": true,
      "widget": "file picker",
      "position": 0
    }, {
      "field": "Proof of address",
      "description": "(Electricity, Gas, Water, Internet, Landline & Mobile Phone Invoice / Payment Schedule) no older than 3 months of the legal representative of your company",
      "type": "File",
      "isRequired": true,
      "position": 1
    }, {
      "field": "Company bank statement",
      "description": "PDF including company name as well as IBAN",
      "type": "File",
      "isRequired": true,
      "position": 2
    }, {
      "field": "Valid proof of ID",
      "description": "ID card or passport if the document has been issued in the EU, EFTA, or EEA / ID card or passport + resident permit or driving licence if the document has been issued outside the EU, EFTA, or EEA of the legal representative of your company",
      "type": "File",
      "isRequired": true,
      "position": 3
    }]
  }, {
    "name": "Whoami",
    "type": "global",
    "endpoint": "/forest/whoami",
    "http_method": "GET"
  }]
}]
```

Reply:
Looking at the file, it reveals that historically some data does not have all the keys. Smart Actions may sometimes not have a "type" for instance.

It would be great to ensure that the data is always formated the same way for collections (attributes/fields/segments/actions/actions.fields).

Reply:
Segments are also formatted like that:
```json
"segments": [
        {
          "id": "Product.Bestsellers",
          "name": "Bestsellers",
          "where": {
          }
        }
      ]
```

and this:
```json
"segments": [{
   "name": "Bestsellers",
}]
```
is enough for the apimap and simpler to read.

Reply:
I did pretty print it. Tell me if that works for you